pytorch_lightning/utilities/parsing.py

```diff
@@ -186,15 +186,17 @@ def lightning_hasattr(model, attribute):
             attr = attribute in model.hparams
         else:
             attr = hasattr(model.hparams, attribute)
+    elif hasattr(model.datamodule, attribute):
+        attr = True
     else:
         attr = False
 
     return attr
 
 
 def lightning_getattr(model, attribute):
```

Comment:
sees very similar logic as above `lightning_hasattr`
can we fist check if it has the attribute and in not ends earlier...

Reply:
I don't know what you mean. Without going through all cases, we won't know if there is an attribute or not.